src/shims/env.rs

```diff
@@ -393,15 +411,21 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let this = self.eval_context_mut();
         this.assert_target_os("windows", "SetCurrentDirectoryW");
 
-        this.check_no_isolation("`SetCurrentDirectoryW`")?;
+        if let IsolatedOp::Reject(reject_with) = this.machine.isolated_op {
+            this.reject_in_isolation("SetCurrentDirectoryW", reject_with)?;
+            let err = Error::new(ErrorKind::NotFound, "rejected due to isolation");
+            this.set_last_error_from_io_error(err)?;
 
-        let path = this.read_path_from_wide_str(this.read_scalar(path_op)?.check_init()?)?;
+            Ok(0)
```

Comment:
Please return early in this branch, so that the code below does not have to be moved to the right.